x/programs/cmd/simulator/src/lib.rs

```diff
@@ -183,7 +183,7 @@ impl<'a> Plan<'a> {
 #[derive(Debug, Deserialize)]
 pub struct BaseResponse {
     /// The numeric id of the step.
-    pub id: u32,
+    pub id: usize,
```

Comment:
I don't know why you would have to cast back and forth, the number is never going to exceed `u32::MAX`